thunder/transforms/quantization.py

```diff
@@ -294,3 +296,230 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+                        requires_grad=requires_grad,
+                    )
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_lora_a, output=proxy_lora_a))
+                    self.check_proxy(
+                        proxy_lora_a,
+                        qs["lora_a.shape"],
+                        thunder.devices.to_device(device),
+                        thunder.dtypes.to_dtype(qs["lora_a.dtype"]),
+                    )
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_lora_b, output=proxy_lora_b))
+                    self.check_proxy(
+                        proxy_lora_b,
+                        qs["lora_b.shape"],
+                        thunder.devices.to_device(device),
+                        thunder.dtypes.to_dtype(qs["lora_b.dtype"]),
+                    )
+
+                    add_trace_output(prologue_trace, proxy_lora_a, subindex=0)
+                    add_trace_output(prologue_trace, proxy_lora_b, subindex=0)
+
+                    new_compute_inputs.append(proxy_lora_a)
+                    new_compute_inputs.append(proxy_lora_b)
+                    # this is not good, because we will have several traces...
+                    additional_proxies[n_lora_a] = proxy_lora_a
+                    additional_proxies[n_lora_b] = proxy_lora_b
+
+        prologue_trace.bound_symbols[-1:-1] = new_bsyms
+        prologue_trace.set_provenance(thunder.core.trace.TraceProvenance("lora linear pass"))
+
+        computation_proxy_map = {
+            csym.name: dict(
+                shape=psym.shape,
+                dtype=psym.dtype,
+            )
+            for psym, csym in zip(prologue_trace.bound_symbols[-1].args[0][0], computation_trace.args)
+            if psym.shape != csym.shape or psym.dtype != csym.dtype
+        }
+
+        new_computation_trace = trace_with_replaced_proxy_metadata(computation_trace, computation_proxy_map)
+        bound_symbols = new_computation_trace.bound_symbols
+        new_computation_trace.bound_symbols = []
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear and bsym.args[1].name in lora_linear_proxies:
+                n = lora_linear_proxies[bsym.args[1].name]
+                with thunder.core.trace.tracectx(computation_trace):
```

Comment:
We should operate on the new computation trace here.
```
with tracectx(new_computation_trace):  # only need this if you don't have it outside somewhere
   lora_a_transpose = thunder.torch.transpose(...)  # this will automatically append a new bsym to the new_computation_trace

Reply:
Yes! Let me update this :)